### PR TITLE
Remote write for span metrics

### PR DESCRIPTION
**Pull request by mapno**
#### PR Checklist

- [x] CHANGELOG updated 
- [x] Documentation added
- [x] Tests updated

### CODE REVIEW

**Review by joe-elliott — COMMENTED** [review]
There is an existing remote write exporter:

https://github.com/open-telemetry/opentelemetry-collector/tree/main/exporter/prometheusremotewriteexporter

Should we use this?

And instead of all of that, is it possible to reuse the remote write that is already present in the agent? @rfratto may have thoughts here.

If we did it would give us wal support as well.

**Comment by mapno** [comment]
Changed the implementation to use a prometheus instance from the Agent to remote write. There are still things missing, but it's a closer look into the approach. The docker-compose example should correctly report span metrics.

File: pkg/tempo/config.go [hunk 1]
```diff
@@ -202,21 +202,17 @@ func (c *RemoteWriteConfig) UnmarshalYAML(unmarshal func(interface{}) error) err
...
 	Dimensions              []spanmetricsprocessor.Dimension `yaml:"dimensions,omitempty"`
-
-	// MetricsExporter is a Prometheus metrics exporter
-	MetricsExporter metricsExporterConfig `yaml:"metrics_exporter,omitempty"`
+	// Namespace if set, exports metrics under the provided value.
+	Namespace string `yaml:"namespace,omitempty"`
+	// ConstLabels are values that are applied for every exported metric.
+	ConstLabels map[string]interface{} `yaml:"const_labels,omitempty"`
+	// Exporter is the configuration to export the metrics
+	Exporter metricsExporterConfig `yaml:"exporter"`
```
Comment (joe-elliott) [inline pkg/tempo/config.go:210]: unless we think we will add more configuration in `metricsExporterConfig` I don't think we should add this extra config level

**Review by joe-elliott — COMMENTED** [review]
I didn't go through all the conversion code, but I'm sure it's flawless. This approach makes sense to me.

File: pkg/tempo/remotewriteexporter/exporter.go [hunk 2]
```diff
@@ -0,0 +1,231 @@
...
+				case pdata.MetricDataTypeDoubleSum, pdata.MetricDataTypeIntSum, pdata.MetricDataTypeDoubleGauge, pdata.MetricDataTypeIntGauge:
+					if err := e.processScalarMetric(app, m); err != nil {
+						return fmt.Errorf("failed to process metric %s", err)
+					}
+				case pdata.MetricDataTypeDoubleHistogram, pdata.MetricDataTypeIntHistogram:
+					if err := e.processHistogramMetrics(app, m); err != nil {
+						return fmt.Errorf("failed to process metric %s", err)
+					}
+				case pdata.MetricDataTypeDoubleSummary:
+					return fmt.Errorf("%s processing unimplemented", m.DataType())
```
Comment (joe-elliott) [inline pkg/tempo/remotewriteexporter/exporter.go:89]: i'm fine with leaving this off for now

File: pkg/tempo/remotewriteexporter/exporter.go [hunk 3]
```diff
@@ -0,0 +1,231 @@
...
+					return fmt.Errorf("unsupported m data type %s", m.DataType())
+				}
+			}
+		}
+	}
+
+	return app.Commit()
+}
+
+func (e *remoteWriteExporter) processHistogramMetrics(app storage.Appender, m pdata.Metric) error {
```
Comment (joe-elliott) [inline pkg/tempo/remotewriteexporter/exporter.go:100]: this is obviously quite complex. i say add some testing and ship it. we can confirm internally that these metrics are in sync with others.

File: pkg/tempo/config.go (hunk 1 above)
Comment (mapno) [reply pkg/tempo/config.go]: Right, the exporter will be configured via the prometheus instance. Removed in 521f7c346f2370b3d71de07e078d7e6d8c84face

File: pkg/tempo/remotewriteexporter/exporter.go (hunk 3 above)
Comment (mapno) [reply pkg/tempo/remotewriteexporter/exporter.go:98]: Added a few unit tests, although I'm not extremely happy with them. I'd like to work something like https://github.com/prometheus/compliance/tree/main/remote_write out to the exporter. Maybe in a future pass?

**Comment by mapno** [comment]
Found an issue while updating the documentation. Agents in scraping mode can't have prometheus instances, so the exporter can't get one to push metrics.

**Comment by joe-elliott** [comment]
Does this PR remove the ability to publish locally? I was thinking we would want both options available. I was also thinking that we need to the ability to log locally to the automatic logging feature.

**Pull request merged**